### DISCUSSION TITLE
Transfer data from Heroku Postgres to Supabase

### DISCUSSION THREAD

**Discussion by ShubhamVerma1811** [supabase/supabase#666] [Questions] [answered]
Hey, I looked for this question but couldn't find.
I have a Postgres DB hosted on Heroku and it has data, is there a way to export that data and import it in Supabase?

**Comment by kiwicopple** [comment, marked as answer]
Definitely the easiest way is to use `pg_dump` (from Heroku) and then `pg_restore` into your Supabase database

https://www.postgresql.org/docs/current/app-pgdump.html

For better performance, it is probably useful to turn off all replication/realtime before you do this:

```
begin; 
  -- remove the realtime publication
  drop publication if exists supabase_realtime; 

  -- re-create the publication but don't enable it for any tables
  create publication supabase_realtime;  
commit;
```

**Reply by flov** [reply]
How can I pg_restore into the supabase database?

**Comment by Rubenburdin** [comment]
Coming to this in 2026 and the pg_dump/pg_restore approach that @kiwicopple mentioned is still the most straightforward way to do a one time data migration from Heroku Postgres to Supabase. flov to answer your question, you can restore into Supabase by running pg_restore --no-owner --no-acl -d "your_supabase_connection_string" your_dump_file using the connection string from your Supabase dashboard under Settings > Database (https://supabase.com/docs/guides/database/connecting-to-postgres). Make sure to use the direct connection string, not the pooler, for bulk operations like restores.

One thing worth mentioning for anyone finding this thread while planning their Heroku migration, if you were also using Heroku Connect to sync Salesforce with your Heroku Postgres, that sync breaks the moment you move off Heroku since Heroku Connect only works within Heroku's ecosystem. A lot of teams discover this too late and end up scrambling to replace it. We've helped quite a few teams through this exact migration at [Stacksync](https://www.stacksync.com/heroku-connect-alternative), (https://supabase.com/partners/integrations/stacksync) replacing Heroku Connect with real time two way Salesforce to Postgres sync that works with Supabase or any Postgres host. No forced hosting, no locked in infrastructure, and it cuts the cost significantly compared to what Heroku Connect charges as your data grows.

For the migration itself, kiwicopple's tip about disabling the realtime publication before the restore is still good advice. Re enable it after the restore completes and then selectively add the tables you actually need realtime on.

Hope this helps anyone going through the same move!
